api/layer/object.go

```diff
@@ -252,31 +259,61 @@ func (n *layer) ListObjectsV2(ctx context.Context, p *ListObjectsParamsV2) (*Lis
 		err        error
 		result     ListObjectsInfoV2
 		allObjects []*ObjectInfo
+		bkt        *BucketInfo
+		accessKey  string
+		cacheKey   string
 	)
 
 	if p.MaxKeys == 0 {
 		return &result, nil
 	}
 
-	if p.ContinuationToken != "" {
-		// find cache with continuation token
+	if bkt, err = n.GetBucketInfo(ctx, p.Bucket); err != nil {
+		return nil, err
+	}
+
+	if data, ok := ctx.Value(api.BoxData).(*accessbox.Box); ok && data != nil {
```

Comment:
Maybe we can somehow reuse `getBoxData` from `put.go` file

Reply:
Moved getBoxData to layer and made it exported because of cycled dependencies issues

Reply:
I suppose now we can change:
```
if data, err := GetBoxData(ctx); data != nil && err == nil {
	accessKey = data.Gate.AccessKey
} else {
	return nil, fmt.Errorf("couldn't get access key from the accessbox: %s", err)
}
cacheKey = createKey(accessKey, bkt.CID)
```
to 
```
if data, err = GetBoxData(ctx); err != nil {
	return nil, err 
}
cacheKey := createKey(data.Gate.AccessKey, bkt.CID)
```